shopware/fastly-meta/6.7/config/fastly/recv/default.vcl

```diff
@@ -60,6 +60,20 @@ if (req.url.path ~ "^/(checkout|account|admin|api|csrf)(/.*)?$") {
     set req.http.x-pass = "1";
 }
 
+# set cache-hash cookie value to header for hashing 1Code has comments. Press enter to view.
+# if header is provided directly the header will take precedence
+if (!req.http.sw-cache-hash) {
+    set req.http.sw-cache-hash = req.http.cookie:sw-cache-hash;
+}
+
+# immediately pass when hash indicates that the content should not be cached
+# note that cache-hash = "not-cacheable" is used to indicate an application state in which the cache should be passed
+# we can not use cache-control headers in that case, as reverse proxies expect to always get the same cache-control headers based on the route
+# dynamically changing the cache-control header is not supported
+if (req.http.sw-cache-hash == "not-cacheable") {
```

Comment:
this sounds evil that u as an client can "ship infrastructure caching" by a cookie value. whats the actual use-case for this?

Reply:
Maybe we document it and leave it commented out  by default so you can activate it when needed?

Reply:
but the diff still has inside it 🤔